projects/media-viewer/src/lib/annotations/services/tags/tags.services.ts

```diff
@@ -0,0 +1,46 @@
+import { Injectable } from '@angular/core';
+import {HttpClient} from '@angular/common/http';
+import {Observable,} from 'rxjs';
+import {TagItemModel} from '../../models/tag-item.model';
+import { v4 as uuid } from 'uuid';
+
+@Injectable()
+export class TagsServices {
+
+  public tagItems: {[id: string]: TagItemModel[]};
+
+  constructor(private http: HttpClient) {}
+
+  public getAllTags(createdBy): Observable<TagItemModel[]> {
+    const url = `/em-anno/tags/${createdBy}`;
+    return this.http.get<TagItemModel[]>(url);
+  }
+
+  getTagItems(annoid): TagItemModel[] {
+    return this.tagItems ? this.tagItems[annoid] : [];
+  }
+
+  updateTagItems(items, annoId) {
+    const snakeCasedWithId = items.map(item => {
+      const id = item.id ? item.id : uuid();
+      return {
+        ...item,
+        id,
+        name: this.snakeCase(item.name)
+      };
+    });
+
+    this.tagItems = {
+      ...this.tagItems,
+      [annoId]: snakeCasedWithId
+    };
+    console.log(this.tagItems);
+  }
+
+  private snakeCase = string => {
+    return string.replace(/\W+/g, " ")
+      .split(/ |\B(?=[A-Z])/)
```

Comment:
not very clear what this is doing. maybe if you put the regex in variables with names that describe what they're for that'll help